nixos/modules/services/networking/unbound.nix

```diff
@@ -6,41 +6,124 @@ let
+          available options.
+        '';
+      };
+      remote-control = mkOption {
+        type = with types; attrsOf validConfigTypes;
+        default = { };
+        description = ''
+          remote-control: clause configuration.
+          See the <citerefentry><refentrytitle>unbound.conf</refentrytitle>
+          <manvolnum>5</manvolnum></citerefentry> manpage for a list of
+          available options.
+        '';
+      };
+      stub-zone = mkOption {
+        type = with types; listOf (attrsOf validConfigTypes);
+        default = [ ];
+        description = ''
+          stub-zone: clause configuration.
+          See the <citerefentry><refentrytitle>unbound.conf</refentrytitle>
+          <manvolnum>5</manvolnum></citerefentry> manpage for a list of
+          available options.
+        '';
+      };
+      forward-zone = mkOption {
+        type = with types; listOf (attrsOf validConfigTypes);
+        default = [ ];
+        example = literalExample ''
+          [
+            {
+              name = ".";
+              forward-addr = "1.1.1.1@853#cloudflare-dns.com";
+            }
+            {
+              name = "example.org.";
+              forward-addr = [
+                "1.1.1.1@853#cloudflare-dns.com"
+                "1.0.0.1@853#cloudflare-dns.com"
+              ];
+            }
+          ];
+        '';
+        description = ''
+          forward-zone: clause configuration.
+          See the <citerefentry><refentrytitle>unbound.conf</refentrytitle>
+          <manvolnum>5</manvolnum></citerefentry> manpage for a list of
+          available options.
+        '';
+      };
+      python = mkOption {
+        type = with types; attrsOf validConfigTypes;
+        default = { };
+        description = ''
+          python: clause configuration.
+          See the <citerefentry><refentrytitle>unbound.conf</refentrytitle>
+          <manvolnum>5</manvolnum></citerefentry> manpage for a list of
+          available options.
+        '';
+      };
+    };
+  };
+
+  yesOrNo = v: if v then "yes" else "no";
+
+  toOption = n: v: "${toString n}: ${v}";
+
+  toConf = n: v:
+    if builtins.isFloat v then (toOption n (toString v))
+    else if isInt v       then (toOption n (toString v))
+    else if isBool v      then (toOption n (yesOrNo v))
+    else if isString v    then (toOption n v)
+    else if isList v      then (concatMapStringsSep "\n  " (toConf n) v)
+    else abort "unbound.toConf: unexpected type (n = ${n})";
+
+  clauseConf = clause: conf: if conf == { } then "" else
+    (concatStringsSep "\n  " (
+      ["${clause}:"] ++ (
+        mapAttrsToList toConf (
+          filterAttrs (_: v: v != null) conf
+        )
+      )));
+
+  confFile = pkgs.writeText "unbound.conf" (concatStringsSep "\n" (flatten [
+    (clauseConf "server" cfg.settings.server)
+    (clauseConf "remote-control" cfg.settings.remote-control)
+    (map (clauseConf "stub-zone") cfg.settings.stub-zone)
```

Comment:
I feel like @Infinisil might have some thoughts on this, possibly regarding `lib.listAsDuplicateKeys`, and how you might only need one option to manage the entire configuration file... :thinking: 

@Infinisil ping ping ping

Reply:
Yeah this here isn't very `settings`-like, as it doesn't support arbitrary clauses (of which apparently there are many according to `man unbound.conf`).

Reply:
Oh yes indeed, it seems I wasn't reading the up-to-date manpage. Will change this right away